URW-7: Navbar Component

corbindonner opened the pull request:
- Link to the ticket [URW-7](https://umnrldev.atlassian.net/browse/URW-7)
- Added 2 pages (Home and About)
- Created Navbar in header component with 4 buttons
- (UMNRL Home, About, Discord, and Twitter)

MDixey17 commented on src/App.tsx:
```diff
@@ -1,7 +1,25 @@
 function App() {
-  return <div className="App">To be implemented</div>;
+  let component;
+  switch (window.location.pathname) {
```
Let's try something different with navigation. I would like you to explore `react-router-dom` and the React router. This is a modern, secure way to navigate between pages. Some resources:
- [My old website that used routing - code base](https://github.com/MDixey17/personal-website/blob/main/src/App.js)
- [W3Schools](https://www.w3schools.com/react/react_router.asp)

corbindonner replied:
I didn't know whether or not to use react-router-dom for external sites like twitch, twitter and discord. I left them as anchor tags for now.

MDixey17 replied:
External links like Twitter, Twitch, Discord, etc. should always be anchor tags, so you are good there

MDixey17 commented on src/components/Navbar/navbar.tsx:
```diff
@@ -0,0 +1,25 @@
+export default function Navbar() {
```
Let's use the arrow syntax as this is more modern practice
`export const Navbar = () => {...}`

MDixey17 commented on src/components/pages/about.tsx:
```diff
@@ -0,0 +1,3 @@
+export default function About() {
+    return <h1>About</h1>
```
Something minor, let's follow a modern file naming convention:
- `.tsx` files should have words capitalize with no spaces (ex: `Navbar.tsx`, `NavbarElements.tsx`, `About.tsx`)
- `.css` files can be camel-case (ex: `globalStyles.css`, `styles.css`)

MDixey17 commented on src/components/pages/about.tsx:
```diff
@@ -0,0 +1,3 @@
+export default function About() {
```
Arrow syntax
`export const About = () => {...}`

MDixey17 commented on src/components/pages/home.tsx:
```diff
@@ -0,0 +1,3 @@
+export default function Home() {
```
Arrow syntax
`export const Home = () => {...}`

MDixey17 reviewed (changes requested):
Solid start, let me run it locally to see if it needs any other changes

corbindonner commented:
- First implementation of react-router-dom's BrowserRouter
- Changed filenames to conform to naming convention
- Changed function signatures to arrow syntax

MDixey17 commented on src/App.tsx:
```diff
@@ -1,7 +1,24 @@
+        <Navbar></Navbar>
+        <Routes>
+          <Route path ="/" element={<Home />} />
+          <Route path="about" element={<About />} />
```
Perfect!

MDixey17 reviewed (approved):
Very nice
